phd-tests/framework/src/guest_os/mod.rs

```diff
@@ -27,6 +27,15 @@ pub(super) enum CommandSequenceEntry<'a> {
     /// Write the specified string as a command to the guest serial console.
     WriteStr(Cow<'a, str>),
 
+    /// Attempt to establish consistent echoing of characters to the guest
+    /// serial console by typing `send` and then waiting for `expect` to appear
+    /// within the supplied `timeout`.
+    EstablishConsistentEcho {
```

Comment:
neat!

is there any particular reason we allow using arbitrary strings for this operation? could we just always use a single character, or is there a reason we might want to do this with multiple strings?

Reply:
The string's arbitrary. It could be just a single character, though if the caller picks something that might already be in the back buffer at the time it requests this operation, it will want to issue a `ClearBuffer` command first to avoid spurious matching. (The way I look at this operation is that it gives guests a way to tell the framework to repeat a pair of `WriteStr`/`WaitFor` commands until the `WaitFor` is satisfied; the flexibility in specifying the sent/expected strings is accidental, not essential, and is mostly there to provide an analogy to the Write/Wait commands.)

Reply:
Ah, yeah, the need to ensure that the string isn't already in the buffer makes sense. :+1: